backend/api/Services/InspectionFindingService.cs

```diff
@@ -13,5 +13,23 @@ public async Task<List<InspectionFinding>> RetrieveInspectionFindings(DateTime l
                                         .ToListAsync();
             return inspectionFindings;
         }
+
+        public async Task<MissionRun?> GetMissionRunByIsarStepId(InspectionFinding inspectionFinding)
+        {
+            return await context.MissionRuns
+                    .Include(missionRun => missionRun.Area).ThenInclude(area => area != null ? area.Plant : null)
+                    .Include(missionRun => missionRun.Robot)
+                    .Where(missionRun => missionRun.Tasks.Any(missionTask => missionTask.Inspections.Any(inspection => inspection.IsarStepId == inspectionFinding.IsarStepId)))
+                    .FirstOrDefaultAsync()
+                    ?? null;
```

Comment:
I'm not sure if the "?? null" does anything here

Reply:
Basically a null check, but ideally should never be null.